File: android/app/src/main/java/hpic/recardify/recardify/Idea.java
package hpic.recardify.recardify;

import androidx.annotation.NonNull;
import java.util.List;
public class Idea {
    public List<String> materialsNeeded, steps, imagesBase64;
    public int timeForCrafting, boxSize, rating, timesRated;
    public String title;
    public Idea() {} // default construct is necessary

    public Idea(String title, int timeForCrafting, int boxSize, int rating, int timesRated, List<String> materialsNeeded, List<String> steps, List<String> imagesBase64) {
        this.title = title;
        this.timeForCrafting = timeForCrafting;
        this.boxSize = boxSize;
        this.rating = rating;
        this.timesRated = timesRated;
        this.steps = steps;
        this.materialsNeeded = materialsNeeded;
        this.imagesBase64 = imagesBase64;
    }

    @NonNull
    public String toString() {
        return this.title + "\n" + this.timeForCrafting + "\n" + this.boxSize + "\n" + this.rating + "\n" + this.timesRated + "\n" + this.materialsNeeded + "\n" + this.steps + "\n" + this.imagesBase64 + "\n";
    }
}

File: lib/idea.dart
class Idea {
    late int timeForCrafting, boxSize, rating, timesRated;
    late String title;
    late List<String> steps, materialsNeeded, imagesBase64;

    Idea({required this.title, required this.timeForCrafting, required this.boxSize, required this.rating, required this.timesRated, required this.materialsNeeded, required this.steps, required this.imagesBase64});

    Map<String, dynamic> toJson() {
        return {
            'title': title,
            'timeForCrafting': timeForCrafting,
            'boxSize': boxSize,
            'rating': rating,
            'timesRated': timesRated,
            'materialsNeeded': materialsNeeded,
            'steps': steps,
            'imagesBase64': imagesBase64
        };
    }

    factory Idea.fromJson(Map<String, dynamic> map) {
        return Idea(
            title: map['title'],
            timeForCrafting: map['timeForCrafting'],
            boxSize: map['boxSize'],
            rating: map['rating'],
            timesRated: map['timesRated'],
            materialsNeeded: List<String>.from(map['materialsNeeded']),
            steps: List<String>.from(map['steps']),
            imagesBase64: List<String>.from(map['imagesBase64'])
        );
    }
}